src/main/java/org/elasticsearch/index/query/TermQueryBuilder.java

```diff
@@ -147,7 +188,66 @@ public void doXContent(XContentBuilder builder, Params params) throws IOExceptio
     }
 
     @Override
+    public Query toQuery(QueryParseContext parseContext) {
+        validate();
+        Query query = null;
+        MapperService.SmartNameFieldMappers smartNameFieldMappers = parseContext.smartFieldMappers(this.fieldName);
+        if (smartNameFieldMappers != null && smartNameFieldMappers.hasMapper()) {
+            query = smartNameFieldMappers.mapper().termQuery(this.value, parseContext);
+        }
+        if (query == null) {
+            query = new TermQuery(new Term(this.fieldName, BytesRefs.toBytesRef(this.value)));
+        }
+        query.setBoost(this.boost);
+        if (this.queryName != null) {
+            parseContext.addNamedQuery(queryName, query);
+        }
+        return query;
+    }
+
+    @Override
+    public void validate() throws ElasticsearchNullPointerException {
+        Preconditions.checkNotNull(this.fieldName, "Fieldname of a TermQuery cannot be null.");
+        Preconditions.checkNotNull(this.value, "Value of a TermQuery cannot be null.");
+    }
+
+    @Override
     protected String parserName() {
         return TermQueryParser.NAME;
     }
+
+    public void readFrom(StreamInput in) throws IOException {
+        fieldName = in.readString();
+        value = in.readGenericValue();
+        boost = in.readFloat();
+        queryName = in.readOptionalString();
+    }
+
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+        out.writeString(fieldName);
+        out.writeGenericValue(value);
+        out.writeFloat(boost);
+        out.writeOptionalString(queryName);
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(fieldName, BytesRefs.toBytesRef(value), boost, queryName);
```

Comment:
Agreed, but this brings me back to my previous problem mentioned in this previous comment https://github.com/elastic/elasticsearch/pull/10669#discussion-diff-29134022. I don't know if it is save parsing the query value using `parser.objectText()` instead of `parser.objectBytes()`, but otherwise the simple equals doesn't work.